File: 0611-valid-triangle-number/0611-valid-triangle-number.cpp
class Solution {
public:
    int triangleNumber(vector<int>& nums) {
        int res = 0;
        sort(nums.begin(), nums.end());
        for (int s1 = 0; s1 < (int)nums.size() - 2; s1++) {
            if(nums[s1] == 0)
            {
                continue;
            }
            int s3 = s1 + 2;
            for(int s2 = s1 + 1; s2 < (int)nums.size() - 1; s2++)
            {
                while(s3 < nums.size() && nums[s3] < nums[s1] + nums[s2])
                {
                    s3++;
                }
                res += s3 - s2 - 1;
            }
        }
        return res;
    }
};

File: 0611-valid-triangle-number/0611-valid-triangle-number.py
class Solution:
    def triangleNumber(self, nums: List[int]) -> int:
        res = 0
        nums = sorted(nums)
        for s1 in range(len(nums) - 2):
            if nums[s1] == 0:
                continue
            s3 = s1 + 2
            print(s3)
            for s2 in range(s1 + 1, len(nums) - 1):
                while s3 < len(nums) and nums[s3] < nums[s1] + nums[s2]:
                    s3 += 1
                res += s3 - s2 - 1
        return res